public/app/plugins/datasource/influxdb/datasource.ts

```diff
@@ -322,6 +322,15 @@ export default class InfluxDatasource extends DataSourceApi<InfluxQuery, InfluxO
       .datasourceRequest(req)
       .then(
         (result: any) => {
+          if (result.data && result.data.results) {
```

Comment:
Do we have some influx db doc link we could add here as a comment so it's easy to find out what kind of errors are these?

Reply:
Unfortunately, not a lot of info has been published there: https://docs.influxdata.com/influxdb/v1.8/tools/api/ - though the most common errors I noticed are about the database or measurement not found, then limits exceeded.

Reply:
Seems like there is a bit more in v2 https://v2.docs.influxdata.com/v2.0/api/#operation/PostQuery so I wonder if we actually support v2 or not.

In any case there is a TS official client https://github.com/influxdata/influxdb-client-js so maybe you can check how do they handle the errors or check if they have response types we could borrow.

Reply:
Yes, there are three types of errors in their client:
1. the generic Error having only `code` and `message` fields,
2. LineProtocolError that extends it with additional `op`, `err` stack and `line` number,
3. LineProtocolLengthError that extends the generic error with `maxLength` property.
But pretty much everything is still being returned in an error message anyway.

Reply:
Yes, there are three types of errors in their client:
1. the generic Error having only `code` and `message` fields,
2. LineProtocolError that extends it with additional `op`, `err` stack and `line` number,
3. LineProtocolLengthError that extends the generic error with `maxLength` property.

But pretty much everything is still being returned in an error message anyway.